Tests/StreamChatUITests/Utils/NukeImageLoader_Tests.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatUI
+import XCTest
+
+final class NukeImageLoader_Tests: XCTestCase {
+    private var requests: [ImageDownloadRequest]!
+    private var imageLoader: MockNukeImageLoader!
+    private var imagePipeline: MockPipeline!
+    
+    override func setUpWithError() throws {
+        imagePipeline = MockPipeline()
+        imageLoader = MockNukeImageLoader(mockPipeline: imagePipeline)
+        requests = (0..<5)
+            .compactMap { URL(string: "https://www.example.com/\($0)") }
+            .map { ImageDownloadRequest(url: $0, options: .init()) }
+    }
+    
+    override func tearDownWithError() throws {
+        imageLoader = nil
+        imagePipeline = nil
+        requests = nil
+    }
+
+    func test_downloadMultipleImages_whenPipelineReturnsResultsInTheSameOrder_thenCompletionResultsAreInTheSameOrderAsPassedInRequests() throws {
+        let expectation = XCTestExpectation()
+        imageLoader.downloadMultipleImages(with: requests) { results in
+            let ids = results.map(MockPipeline.pipelineLoadingCompletionIndex(for:))
+            XCTAssertEqual(ids, [1, 2, 3, 4, 5])
+            expectation.fulfill()
+        }
+        imagePipeline.dispatchPipelineCompletions(order: .same)
+        wait(for: [expectation])
+    }
+    
+    func test_downloadMultipleImages_whenPipelineReturnsResultsInReverseOrder_thenCompletionResultsAreInTheSameOrderAsPassedInRequests() throws {
+        let expectation = XCTestExpectation()
+        imageLoader.downloadMultipleImages(with: requests) { results in
+            let ids = results.map(MockPipeline.pipelineLoadingCompletionIndex(for:))
+            XCTAssertEqual(ids, [1, 2, 3, 4, 5])
+            expectation.fulfill()
+        }
+        imagePipeline.dispatchPipelineCompletions(order: .reversed)
+        wait(for: [expectation])
+    }
+    
+    func test_downloadMultipleImages_whenPipelineReturnsResultsInShuffledOrder_thenCompletionResultsAreInTheSameOrderAsPassedInRequests() throws {
+        let expectation = XCTestExpectation()
+        imageLoader.downloadMultipleImages(with: requests) { results in
+            let ids = results.map(MockPipeline.pipelineLoadingCompletionIndex(for:))
+            XCTAssertEqual(ids, [1, 2, 3, 4, 5])
+            expectation.fulfill()
+        }
+        imagePipeline.dispatchPipelineCompletions(order: .shuffled)
+        wait(for: [expectation])
```

Comment:
This looks like is failing on Xcode 14 build 🤔 Ideally we want to use the "DefaultTimeout" that is part of our test tools.

Reply:
Thank you!